runelite-client/src/main/java/net/runelite/client/plugins/microbot/blackjack/BlackJackScript.java

```diff
@@ -0,0 +1,612 @@
+import net.runelite.api.coords.WorldArea;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.api.events.HitsplatApplied;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.plugins.microbot.blackjack.enums.State;
+import net.runelite.client.plugins.microbot.Microbot;
+import net.runelite.client.plugins.microbot.Script;
+import net.runelite.client.plugins.microbot.util.bank.Rs2Bank;
+import net.runelite.client.plugins.microbot.util.combat.Rs2Combat;
+import net.runelite.client.plugins.microbot.util.equipment.Rs2Equipment;
+import net.runelite.client.plugins.microbot.util.gameobject.Rs2GameObject;
+import net.runelite.client.plugins.microbot.util.inventory.Rs2Inventory;
+import net.runelite.client.plugins.microbot.util.keyboard.Rs2Keyboard;
+import net.runelite.client.plugins.microbot.util.npc.Rs2Npc;
+import net.runelite.client.plugins.microbot.util.player.Rs2Player;
+import net.runelite.client.plugins.microbot.util.shop.Rs2Shop;
+import net.runelite.client.plugins.microbot.util.walker.Rs2Walker;
+import net.runelite.client.plugins.microbot.util.widget.Rs2Widget;
+
+import java.awt.event.KeyEvent;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static net.runelite.client.plugins.microbot.blackjack.enums.State.*;
+import static net.runelite.client.plugins.microbot.util.Global.sleepUntilTrue;
+import static net.runelite.client.plugins.microbot.util.math.Random.random;
+import static net.runelite.client.plugins.microbot.util.walker.Rs2Walker.getTile;
+
+
+public class BlackJackScript extends Script {
+    public static double version = 1.1;
+    public static State state = BANKING;
+    BlackJackConfig config;
+    boolean firstPlayerOpenCurtain = false;
+    boolean initScript = false;
+    boolean npcIsTrapped = false;
+    boolean knockout = false;
+    static boolean isPlayerNearby = false;
+    static boolean npcsCanSeeEachother = false;
+    NPC npc;
+    public static List<NPC> npcsInArea = new ArrayList();
+    int g = 0;
+    static int playerHit=0;
+    int previousHP;
+    int xpDrop;
+    int bjXPDrop;
+    int bjCycle = 0;
+    int emptyJug = 1935;
+    int notedWine = 1994;
+    int unnotedWine= 1993;
+    int pollniveachTeleport = 11743;
+    long xpdropstartTime;
+    long startTime;
+    long endTime;
+    long previousAction;
+    long followingAction;
+    WorldArea shopsArea = new WorldArea(3359,2988,4,4,0);
+    WorldPoint shopsLocation = new WorldPoint(3359, 2988, 0);
+    private boolean hasRequiredItems() {
+        return Rs2Inventory.hasItem("Coins")
+                && Rs2Equipment.isWearing("blackjack")
+                && (Rs2Equipment.isWearing(config.teleportItemToBank()) || Rs2Inventory.hasItem(config.teleportItemToBank()))
+                && Rs2Inventory.hasItem(notedWine);
+    }
+    private boolean withdrawRequiredItems() {
+        sleepUntil(() -> Rs2Bank.isOpen());
```

Comment:
You already check if the bank is open when calling this method, this sleepUntil seems redundant